Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py

```diff
@@ -22,6 +22,42 @@
 # plugin architecture.
 #
 
+class TagValueList(list):
```

Comment:
This looks interesting, elegant, Pythonic but way too complicated for what it does. I usually prefer optimizing for simplicity and readability instead, because those are the most important in the long term. This code looks also potentially slow, which is quite important, because for a single study, these checks are done for thousands of frames, each frame contains several fields that are compared, and  a field may contain vectors with 6 values. I don't want to ruin this elegent Pythonic design, but if there are objective disadvantages (especially slow speed) then we may consider reverting to the previous more direct implementation.

Reply:
yes, I thought I had made basically that same comment but maybe I didn't finish the comment before closing the browser tab.  I find the use of an extra class here clever, but I don't see it improving readability and I don't see any case were we would want to reuse this code.  The speed issue @lassoan raises is also a good point.

Reply:
@jcfr @lassoan since also the issues listed in  https://github.com/Slicer/Slicer/pull/7362#discussion_r1388508506 

for the moment I will revert to 
https://github.com/Slicer/Slicer/commit/64b037eb264de0d6f87c4b0aa79e49bd399176ca

Then up to you guys how to proceed

Reply:
done, please have a look at https://github.com/Slicer/Slicer/commit/59c63ff45c65291973fd4d343fc53d5a19c6ceaa

Reply:
Considering that the `numpy.ndarray` is not a hashable type, this prevent us from indexing into the `subseriesFiles` map, the original implementation ends up being more straightforward.